src/components/data/data-utils.js

```diff
@@ -949,72 +948,91 @@ export const SingleEntityEditor = ({ entityType, entityName, attributeName, attr
   ])
 }
 
-export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames, entityTypes, workspaceId: { namespace, name }, onDismiss, onSuccess }) => {
+const editEntitiesSetValue = ({ namespace, name }, entityType, entities, attributeToEdit, newValue) => {
+  const entityUpdates = _.map(entity => ({
+    entityType,
+    name: entity.name,
+    operations: [{
+      op: 'AddUpdateAttribute',
+      attributeName: attributeToEdit,
+      addUpdateAttribute: prepareAttributeForUpload(newValue)
+    }]
+  }), entities)
+
+  return Ajax()
+    .Workspaces
+    .workspace(namespace, name)
+    .upsertEntities(entityUpdates)
+}
+
+const editEntitiesConvertType = ({ namespace, name }, entityType, entities, attributeToEdit, newType) => {
+  const { type, entityType: referenceEntityType } = newType
+
+  const entityUpdates = _.map(entity => ({
+    entityType,
+    name: entity.name,
+    operations: [{
+      op: 'AddUpdateAttribute',
+      attributeName: attributeToEdit,
+      addUpdateAttribute: convertAttributeValue(entity.attributes[attributeToEdit], type, referenceEntityType)
+    }]
+  }), entities)
+
+  return Ajax()
+    .Workspaces
+    .workspace(namespace, name)
+    .upsertEntities(entityUpdates)
+}
```

Comment:
There appears to be quite a bit of code duplication between `editEntitiesSetValue` and `editEntitiesConvertType` and they are used only in `saveAttributeEdits`. Have you considered inlining them in `saveAttributeEdits`?

Reply:
Done in [7435e64](https://github.com/DataBiosphere/terra-ui/pull/3313/commits/7435e6447778474f733b7f1ae3cb8bf9d15c1f5b).